libshaderc_util/src/spirv_tools_wrapper.cc

```diff
@@ -79,4 +81,22 @@ bool SpirvToolsAssemble(const string_piece assembly, spv_binary* binary,
   return result;
 }
 
+bool SpirvToolsOptimize(const EnableOptimizationPass& enabled_passes,
+                        std::vector<uint32_t>* binary, std::string* errors) {
+  using namespace spvtools;
+
+  errors->clear();
+
+  spvtools::Optimizer optimizer(SPV_ENV_VULKAN_1_0);
+  std::ostringstream oss;
+  optimizer.SetMessageConsumer(
+      [errors](spv_message_level_t, const char*, const spv_position_t&,
+               const char* message) { *errors = message; });
```

Comment:
Right. Actually the previous impl. will just retain the last message. With `*oss <<` we actually have them all.